shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -521,6 +618,27 @@ public FlutterShellArgs getFlutterShellArgs() {
     return FlutterShellArgs.fromIntent(getIntent());
   }
 
+  /**
+   * Returns the ID of a statically cached {@link FlutterEngine} to use within this
+   * {@code FlutterActivity}, or {@code null} if this {@code FlutterActivity} does not want to
+   * use a cached {@link FlutterEngine}.
+   */
+  @Override
+  @Nullable
+  public String getCachedEngineId() {
+    return getIntent().getStringExtra(EXTRA_CACHED_ENGINE_ID);
```

Comment:
I forget how this works. If you made an activity with an Intent and rotate the screen, it's the same intent instance?

Reply:
I don't know about the `Intent` instance, but the `Intent` should report the same extras.

Reply:
SG